elide-core/src/main/java/com/yahoo/elide/core/security/executors/AggregationStorePermissionExecutor.java

```diff
@@ -31,13 +31,39 @@ public AggregationStorePermissionExecutor(RequestScope requestScope) {
         super(log, requestScope);
     }
 
+    /**
+     * Checks user checks for the requested fields.
+     * expression = (field1Rule OR field2Rule ... OR fieldNRule)
+     * @param annotationClass annotation class
+     * @param resource resource
+     * @param requestedFields the list of requested fields
+     * @param <A>
+     * @return ExpressionResult - result of the above any field expression
+     */
     @Override
     public <A extends Annotation> ExpressionResult checkPermission(Class<A> annotationClass,
                                                                    PersistentResource resource,
                                                                    Set<String> requestedFields) {
-        return ExpressionResult.PASS;
+        Expression expression = expressionBuilder.buildUserCheckAnyExpression(
```

Comment:
Can we add a Preconditions check or maybe return FAIL if annotationClass is anything other than a ReadPermission for all of these checks?